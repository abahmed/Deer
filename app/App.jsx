import React, {Component} from 'react';

import AddNote from './components/add-note';
import NoteList from './containers/note-list';

class App extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div>
        <h1> Deer </h1>
        <AddNote />
        <NoteList />
      </div>
    )
  }
}

export default App
