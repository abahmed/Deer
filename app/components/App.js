import React from 'react'
import { Switch, Route, BrowserRouter, Redirect } from 'react-router-dom'
import routes from '../routes'

export default () => (
  <div>
    <BrowserRouter>
      <Switch>
        {routes.map((route, index) => (
          <Route
            key={index}
            path={route.path}
            exact={route.exact}
            component={route.component}
          />
        ))}
        <Redirect to='/' />
      </Switch>
    </BrowserRouter>
  </div>
)
