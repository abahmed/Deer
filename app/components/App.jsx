import React from 'react'
import { Switch, Route, Redirect, HashRouter } from 'react-router-dom'
import routes from '../routes'

export default () => (
  <HashRouter>
    <Switch>
      {routes.map((route, index) => (
        <Route
          key={index}
          path={route.path}
          exact={route.exact}
          component={route.component}
        />
      ))}
      <Redirect to='/' />
    </Switch>
  </HashRouter>
)
