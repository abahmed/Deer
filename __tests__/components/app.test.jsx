import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from '../../app/components/App'

jest.mock('../../__mocks__/electron')

describe('<App />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<App />)
      const component = wrapper.dive()

      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
