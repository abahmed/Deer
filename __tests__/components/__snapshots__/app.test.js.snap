// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="0"
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      key="1"
      path="/welcome"
    />
    <Redirect
      push={false}
      to="/"
    />
  </Switch>
</Router>
`;
